src/commando/core.py

```diff
@@ -1,2 +1,79 @@
-def hello_world():
-    return "Hello, world!"
+import subprocess
+import tempfile
+from functools import partial
+from subprocess import PIPE, STDOUT, CalledProcessError
+
+
+def _pprint(repo, message):
+    print(f"{repo}> {message}")
+
+
+def _run(cwd, results, command, **kwargs):
+    result = subprocess.run(  # pylint:disable=subprocess-run-check
+        command, cwd=cwd, stdout=PIPE, stderr=STDOUT, text=True, **kwargs
+    )
+    results.append(result)
+    result.check_returncode()
+    return result
+
+
+def _process_repo(
+    repo, command, commit_message, branch, pr_title, pr_body
+):  # pylint:disable=too-many-arguments
+
+    # Collect the results of all commands run for this repo.
+    results = []
+
+    pprint = partial(_pprint, repo)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        run = partial(_run, tmpdirname, results)
+
+        try:
+            pprint("cloning")
+            run(["gh", "repo", "clone", repo, tmpdirname, "--", "--quiet"])
+
+            pprint(command)
+            run(command, shell=True)
+
+            if run(["git", "status", "--porcelain"]).stdout:
+                pprint("committing changes")
+                run(["git", "switch", "--force-create", branch, "origin/main"])
+                run(["git", "add", "--all"])
+                run(["git", "commit", "--message", commit_message])
+
+                pprint("sending PR")
+
+                try:
+                    pprint(
+                        run(
+                            ["gh-pr-upsert", "--title", pr_title, "--body", pr_body]
+                        ).stdout.rstrip()
```

Comment:
It's a bit weird for commando to be calling gh-pr-upsert (which is another Python package) as a subprocess. But this enables command to use gh-pr-upsert's CLI which is where gh-pr-upsert's defaults for various command line options are currently implemented. It also means that gh-pr-upsert accepts the `cwd` argument (from the `_run()` method above that runs it) which is the directory that the currently-being-processed repo has been cloned to.

To enable commando to use gh-pr-upsert as a Python library rather than as a subprocess we'd have to add a Python interface to gh-pr-upsert that implements its default values and also accepts a CWD argument.